### PR TITLE
Allow workspace name length up to 100 chars

### LINKED ISSUE #6130: Allowed workspace name length should be up to 100 chars (currently only 20 is allowed)
Related osio issue - https://github.com/openshiftio/openshift.io/issues/446
On osio there is a requirement with prefixing workspace name with github repository name. Currently, workspace name is validated against the following pattern: `"[a-zA-Z0-9][-_.a-zA-Z0-9]{1,18}[a-zA-Z0-9]"` which allows names up to 20 chars and GitHub repo name length can be up to 100 chars. So, with current validation it would not be possible to create workspaces with names based on github repos

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
Increase limits on server-side and in UD.

### What issues does this PR fix or reference?
Fixes #6130 

#### Release Notes
Allowed workspace name length up to 100 chars

### CODE REVIEW

**Comment by benoitf** [comment]
@slemeur : what about UI when we have workspace with 100 characters

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
 public class DefaultWorkspaceValidator implements WorkspaceValidator {
   /* should contain [3, 20] characters, first and last character is letter or digit, available characters {A-Za-z0-9.-_}*/
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java]: Looks like comment has outdated value `[3, 20 ] characters`

**Comment by garagatyi** [comment]
@benoitf @slemeur we can use another limit in UD I think. It doesn't have to be equal to the server-side limit.

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java (hunk 1 above)
Comment (garagatyi) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java]: ))) yep

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java [hunk 2]
```diff
@@ -32,9 +32,9 @@
  */
 @Singleton
 public class DefaultWorkspaceValidator implements WorkspaceValidator {
-  /* should contain [3, 20] characters, first and last character is letter or digit, available characters {A-Za-z0-9.-_}*/
+  /* should contain [3, 100] characters, first and last character is letter or digit, available characters {A-Za-z0-9.-_}*/
   private static final Pattern WS_NAME =
-      Pattern.compile("[a-zA-Z0-9][-_.a-zA-Z0-9]{1,18}[a-zA-Z0-9]");
+      Pattern.compile("[a-zA-Z0-9][-_.a-zA-Z0-9]{1,98}[a-zA-Z0-9]");
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java:37]: What about fixes for corresponding tests like [this one](https://github.com/garagatyi/che/blob/wsName100/wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidatorTest.java#L88)?

Comment (garagatyi) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceValidator.java:37]: nice catch! Fixed

**Review by sleshchenko — APPROVED** [review]

**Comment by slemeur** [comment]
@garagatyi + @ashumilova : How is it looking in the dashboard? 
I'm worry mostly about workspace list and sidebar navigation?
Are we truncating them with "..." + tooltip on hover?

**Comment by garagatyi** [comment]
@slemeur I'll add screenshots. Do you have any specific cases in mind? Like resolution, browser, OS, window size?

**Comment by garagatyi** [comment]
Name is cut:
image
image
image
In the next screenshot 20-chars long name is cut too
image

**Comment by slemeur** [comment]
so @ashumilova , @olexii4 @akurinnoy - we need to handle that better.

**Comment by ashumilova** [comment]
okay, we can review this page with such resolution.

**Review by ashumilova — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

**Comment by ashumilova** [comment]
Issue for dashboard team : https://github.com/eclipse/che/issues/6621

**Comment by garagatyi** [comment]
Waiting for resolution of #6621 to merge this PR

**Comment by garagatyi** [comment]
@ashumilova Since adapting to long names is implemented should I do any changes in this PR or I can merge?

**Review by benoitf — APPROVED** [review]

**Comment by ashumilova** [comment]
@garagatyi you need to be aligned with master branch

**Comment by garagatyi** [comment]
@slemeur Can you add your review to the PR now?

**Review by slemeur — APPROVED** [review]
LGTM

**Pull request merged**